cluster/gce/util.sh

```diff
@@ -154,11 +154,23 @@ function copy-to-staging() {
   local -r gs_url=$2
   local -r tar=$3
   local -r hash=$4
+  local -r basename_tar=$(basename ${tar})
+
+  #check whether this tar alread exists and has the same hash
+  #if it matches, then don't bother uploading it again
+  local -r remote_tar_hash=$(gsutil hash -h -m ${staging_path}/${basename_tar} 2>/dev/null | grep "Hash (md5):" | awk -F ':' '{print $2}')
```

Comment:
In this particular case the hash is operating on the remote file, and I tested that it works.

Looks like that documentation you referenced is out of date, and this feature was added to support hash on cloud files in gsutil v4.21. Here is the related issue for tracking hash of cloud files - https://github.com/GoogleCloudPlatform/gsutil/issues/369